src/Features/CSharp/Portable/CodeRefactorings/PullMemberUp/Dialog/CSharpInterfacePullerViaDialog.cs

```diff
@@ -0,0 +1,53 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Composition;
+using Microsoft.CodeAnalysis.CodeGeneration;
+using Microsoft.CodeAnalysis.CodeRefactorings.PullMemberUp.Dialog;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Editing;
+using Microsoft.CodeAnalysis.Host.Mef;
+
+namespace Microsoft.CodeAnalysis.CSharp.CodeRefactorings.PullMemberUp.Dialog
+{
+    [ExportLanguageService(typeof(AbstractInterfacePullerWithDialog), LanguageNames.CSharp), Shared]
+    internal class CSharpInterfacePullerViaDialog : AbstractInterfacePullerWithDialog
+    {
+        protected override void ChangeMemberToPublicAndNonStatic(DocumentEditor editor, ISymbol symbol, SyntaxNode memberSyntax, SyntaxNode containingTypeNode, ICodeGenerationService codeGenerationService)
+        {
+            var modifier = DeclarationModifiers.From(symbol).WithIsStatic(false);
+            if (symbol is IEventSymbol eventSymbol)
+            {
+                if (memberSyntax.Parent != null &&
+                    memberSyntax.Parent.Parent is BaseFieldDeclarationSyntax eventDeclaration)
+                {
+                    if (eventDeclaration.Declaration.Variables.Count > 1)
+                    {
+                        var options = new CodeGenerationOptions(generateMethodBodies: false, generateMembers: false);
+                        var publicAndNonStaticSymbol = CodeGenerationSymbolFactory.CreateEventSymbol(
+                            eventSymbol.GetAttributes(),
+                            Accessibility.Public,
+                            modifier,
+                            eventSymbol.Type,
+                            eventSymbol.ExplicitInterfaceImplementations,
+                            eventSymbol.Name,
+                            eventSymbol.AddMethod,
+                            eventSymbol.RemoveMethod,
+                            eventSymbol.RaiseMethod);
+                        var nonStaticSyntax = codeGenerationService.CreateEventDeclaration(publicAndNonStaticSymbol, options: options);
+                        editor.RemoveNode(memberSyntax);
+                        editor.AddMember(containingTypeNode, nonStaticSyntax);
```

Comment:
i don't actually understand what's going on (and that indicates this code needs comments).  Specifically, we only get here if there are multiple event variables.  But you then remove the entire member (i.e. all the event variables) and seem to be replacing it with just a single member.  That feels lke we're losing variables here.

Reply:
I will add comments and cut it into small methods